src/naive/date.rs

```diff
@@ -1903,7 +1751,7 @@ impl DoubleEndedIterator for NaiveDateDaysIterator {
             return None;
         }
         let current = self.value;
-        self.value = current.pred();
+        self.value = current.pred_opt().unwrap();
```

Comment:
could we use `self.value = current.pred_opt()?;` and then remove the whole if block?